src/service.rs

```diff
@@ -220,6 +240,21 @@ impl GrpcErrResponse {
     }
 }
 
+impl From<DeniedHttpResponse> for DirectResponse {
```

Comment:
I'd move this impl to the "auth" module. This (looking at the `use`s) is already a little too coupled to impl. details, which probably should all be cleanup too.